src/models/filter.rs

```diff
@@ -196,19 +196,32 @@ impl SourceCodeUnit {
     // This ensures that the below while loop considers the current node too when checking for filters.
     // It does not make sense to check for filter if current node is a "leaf" node.
     let mut initial_node = node;
-    if node.child_count() > 0 {
-      initial_node = node.child(0).unwrap();
-    }
 
     // No enclosing node is provided
     if filter.enclosing_node().get_query().as_str() == DEFAULT_ENCLOSING_QUERY {
```

Comment:
should we make `enclosing_node: Optional[str]` ? Also let's update the `.pyi` file too then

Reply:
I think it;s already optional in the pyi. Must have done this in the previous PR

Reply:
i see. 
Let's update this `#pynew` too 
https://github.com/uber/piranha/blob/master/src/models/filter.rs#L83 to use `default_enclosing_node`

And in pyi instead of `enclosing_node: str = ''` lets make it `enclosing_node: Optional[str] = None` Some sense I want to keep the signature of `#pynew` and `pyi` consistent (As much as we can)

(I know its logically still the same, but just lets reflect our intention in code)